assets/app/scripts/services/logs/logLinks.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .factory('logLinks', [
+    '$anchorScroll',
+    '$document',
+    '$location',
+    '$window',
+    function($anchorScroll, $document, $location, $window) {
+      var doc = $document[0];
+      return {
+        makeDownload: function(obj) {
+          var a = doc.createElement('a');
+          a.href = $window.URL.createObjectURL(new Blob([obj], {type: 'text/plain'}));
+          a.download = 'log.txt';
+          doc.body.appendChild(a);
+          a.click();
+          doc.body.removeChild(a);
+        },
```

Comment:
After reading the doc, the main thing that concerns me is that it doesn't garbage collect those object URLs until the document is unloaded unless you explicitly do a revokeObjectURL call.  Which I think means its keeping a whole extra copy of the log in memory every time you call createObjectURL.  

If we keep this then I think we would need to limit it to only ever creating one per download link (clicking multiple times shouldn't result in multiple objects), and revoke that object URL when we navigate away from the page it was on.  

The download link shouldn't show up at all on browsers where this isn't supported, although from a cursory glance it seems like all the ones we support have it in some form.